File: native/ios/LokinLocationShellInstaller.swift
import Foundation
import WebKit

/// Retain one instance for the lifetime of the trusted LOKIN WKWebView.
/// This is the only wiring the exported iOS shell needs after adding the local
/// LokinLocationCore Swift package and merging the location Info.plist keys.
@MainActor
final class LokinLocationShellInstaller {
    private let bridge: LokinLocationBridge

    init(webView: WKWebView, allowedHost: String = "lokin-ai-app-604c3139.base44.app") {
        self.bridge = LokinLocationBridge(webView: webView, allowedHost: allowedHost)
        self.bridge.install()
    }

    deinit {
        bridge.uninstall()
    }
}

File: native/android/LokinLocationShellInstaller.kt
package ai.lokin.location

import android.app.Activity
import android.webkit.WebView

/**
 * Retain one instance for the lifetime of the trusted LOKIN WebView.
 * The exported shell calls install() after WebView creation and destroy() when
 * the Activity/WebView is torn down.
 */
class LokinLocationShellInstaller(
    activity: Activity,
    private val webView: WebView,
    allowedHost: String = "lokin-ai-app-604c3139.base44.app"
) {
    private val bridge = LokinLocationBridge(activity, webView, allowedHost)

    fun install() {
        webView.addJavascriptInterface(bridge, "LokinLocation")
    }

    fun destroy() {
        webView.removeJavascriptInterface("LokinLocation")
        bridge.destroy()
    }
}
